DSharpPlus/Clients/DiscordClient.Dispatch.cs

```diff
@@ -1589,6 +1677,69 @@ internal async Task OnMessageReactionRemoveEmojiAsync(ulong messageId, ulong cha
+        {
+            stage.Discord = this;
+
+            await this._stageInstanceCreated.InvokeAsync(this, new StageInstanceCreateEventArgs { StageInstance = stage }).ConfigureAwait(false);
+        }
+
+        internal async Task OnStageInstanceUpdateEventAsync(DiscordStageInstance stage)
+        {
+            stage.Discord = this;
+
+            await this._stageInstanceUpdated.InvokeAsync(this, new StageInstanceUpdateEventArgs { StageInstance = stage }).ConfigureAwait(false);
+        }
+
+        internal async Task OnStageInstanceDeleteEventAsync(DiscordStageInstance stage)
+        {
+            stage.Discord = this;
+
+            await this._stageInstanceDeleted.InvokeAsync(this, new StageInstanceDeleteEventArgs { StageInstance = stage }).ConfigureAwait(false);
+        }
+
+        #endregion
+
+        #region Thread
+
+        internal async Task OnThreadCreateEventAsync(DiscordThreadChannel thread)
+        {
+            thread.Discord = this;
+
+            await this._threadCreated.InvokeAsync(this, new ThreadCreateEventArgs { ThreadChannel = thread }).ConfigureAwait(false);
+        }
+
+        internal async Task OnThreadUpdateEventAsync(DiscordThreadChannel thread)
+        {
+            thread.Discord = this;
+
+            await this._threadUpdated.InvokeAsync(this, new ThreadUpdateEventArgs { ThreadChannel = thread }).ConfigureAwait(false);
+        }
+
+        internal async Task OnThreadDeleteEventAsync(DiscordThreadChannel thread)
+        {
+            thread.Discord = this;
+
+            await this._threadDeleted.InvokeAsync(this, new ThreadDeleteEventArgs { Id = thread.Id, GuildId = thread.GuildId, ParentChannelId = thread.ParentChannelId, Type = thread.Type }).ConfigureAwait(false);
+        }
+
+        internal async Task OnThreadListSyncEventAsync(ulong guild_id, IEnumerable<ulong?> channel_ids, IEnumerable<DiscordThreadChannel> threads, IEnumerable<DiscordThreadChannelMember> members)
+            => await this._threadListSynced.InvokeAsync(this, new ThreadListSyncEventArgs { GuildId = guild_id, ChannelIds = channel_ids, Threads = threads, Members = members }).ConfigureAwait(false);
+
+        internal async Task OnThreadMemberUpdateEventAsync(DiscordThreadChannelMember member)
+        {
+            member.Discord = this;
+
+            await this._threadMemberUpdated.InvokeAsync(this, new ThreadMemberUpdateEventArgs { ThreadMember = member }).ConfigureAwait(false);
+        }
+
+        internal async Task OnThreadMemberUpdatesEventAsync(ulong id, ulong guild_id, int member_count, IEnumerable<DiscordThreadChannelMember> addedMembers, IEnumerable<ulong?> removed_member_ids)
+            => await this._threadMembersUpdated.InvokeAsync(this, new ThreadMembersUpdateEventArgs { Id = id, GuildId = guild_id, MemberCount = member_count, AddedMembers = addedMembers, RemovedMemberIds = removed_member_ids }).ConfigureAwait(false);
```

Comment:
Use concrete DiscordGuild objects from the cache, don't pass an ID

Reply:
Uses concrete DiscordGuild. There is no cache for threads yet so still passing id for that.